src/integration/blockchain/ain/services/crypto.service.ts

```diff
@@ -9,30 +9,30 @@ import { Config } from 'src/config/config';
 @Injectable()
 export class CryptoService {
   public verifySignature(message: string, address: string, signature: string): boolean {
-    const blockchains = this.getBlockchainsBasedOn(address);
+    const blockchain = this.getBlockchainsBasedOn(address);
 
     let isValid = false;
     try {
-      isValid = this.verify(message, address, signature, blockchains);
+      isValid = this.verify(message, address, signature, blockchain);
     } catch (e) {}
 
-    if (!isValid && !blockchains.includes(Blockchain.ETHEREUM)) {
-      isValid = this.fallbackVerify(message, address, signature, blockchains);
+    if (!isValid && !blockchain.includes(Blockchain.ETHEREUM)) {
+      isValid = this.fallbackVerify(message, address, signature, blockchain);
     }
     return isValid;
   }
 
-  public getBlockchainsBasedOn(address: string): Blockchain[] {
-    if (isEthereumAddress(address)) return [Blockchain.ETHEREUM, Blockchain.BINANCE_SMART_CHAIN];
-    if (this.isBitcoinAddress(address)) return [Blockchain.BITCOIN];
-    return [Blockchain.DEFICHAIN];
+  public getBlockchainsBasedOn(address: string): Blockchain {
+    if (isEthereumAddress(address)) return Blockchain.ETHEREUM;
+    if (this.isBitcoinAddress(address)) return Blockchain.BITCOIN;
+    return Blockchain.DEFICHAIN;
   }
```

Comment:
I would revert this change, because the address is still valid on multiple chains. Maybe add a second method for the default chain?
```ts
public getDefaultBlockchainBasedOn(address: string): Blockchain {
   return this.getBlockchainsBasedOn(address)[0];
}